fhir-bucket/src/main/java/com/ibm/fhir/bucket/reindex/ClientDrivenReindexOperation.java

```diff
@@ -298,30 +370,36 @@ private boolean extractIndexIds(Parameters output) {
      * Thread to repeatedly call the $reindex operation until done or error.
      */
     private void callReindexOperationInLoop() {
-        while (this.running && this.active) {
-            String indexIds = getIndexIdsToReindex();
-            if (!indexIds.isEmpty()) {
-                boolean ok = false;
-                try {
-                    ok = callReindexOperation(indexIds);
-                } catch (DataAccessException x) {
-                    // allow active be set to false.  This will notify monitorLoop something is wrong.
-                    // Probably all threads will encounter the same exception and monitorLoop will
-                    // try to refill the pool if all threads exit.
-                    logger.severe("DataAccessException caught when contacting FHIR server. FHIR client thread will exit." + x.toString() );
-                } catch (IllegalStateException x) {
-                    // Fail for this exception too. fhir-bucket fhir client suggests this exception results from config error.
-                    // So probably this will be caught first time monitorLoop calls callOnce and not here.
-                    logger.severe("IllegalStateException caught. FHIR client thread will exit." + x.toString() );
-                }
-                if (!ok) {
-                    // stop everything on the first failure
-                    this.active = false;
+        try {
+            while (this.running && this.active) {
+                String indexIds = getIndexIdsToReindex();
+                if (!indexIds.isEmpty()) {
+                    boolean ok = false;
+                    try {
+                        ok = callReindexOperation(indexIds);
+                    } catch (DataAccessException x) {
+                        // allow active be set to false.  This will notify monitorLoop something is wrong.
+                        // Probably all threads will encounter the same exception and monitorLoop will
+                        // try to refill the pool if all threads exit.
+                        logger.severe("DataAccessException caught when contacting FHIR server. FHIR client thread will exit." + x.toString() );
+                    } catch (IllegalStateException x) {
+                        // Fail for this exception too. fhir-bucket fhir client suggests this exception results from config error.
+                        // So probably this will be caught first time monitorLoop calls callOnce and not here.
+                        logger.severe("IllegalStateException caught. FHIR client thread will exit." + x.toString() );
+                    }
+                    if (!ok) {
+                        // stop everything on the first failure
+                        this.active = false;
+                    }
                 }
             }
+        } catch (Throwable t) {
+            logger.severe("Throwable caught. FHIR client thread will exit." + t.toString() );
```

Comment:
does Throwable.toString print the stacktrace?
maybe better to do this?
```suggestion
            logger.log("Throwable caught. FHIR client thread will exit.", t);
```

Reply:
Updated.